src/cobovault.js

```diff
@@ -0,0 +1,313 @@
+                total,
+                workloads: [],
+                result: decodeUR(enhancedQRs),
+                type: "ur"
+            };
+        } else {
+            return {
+                success: false,
+                current: enhancedQRs.length,
+                total,
+                workloads: enhancedQRs,
+                result: "",
+                type: "ur"
+            };
+        }
+    } else {
+        return {
+            success: false,
+            current: 0,
+            total: 0,
+            workloads: [],
+            result: "",
+            type: "ur"
+        };
+    }
+}
+
+function commandMessage(data) {
+    return {
+        ...{
+            state: PENDING,
+            level: INFO,
+            code: "cobo.command",
+            mode: "wallet",
+        },
+        ...{text: `${data.instructions}`},
+        ...data,
+    };
+}
+
+/**
+ * Base class for interactions with CoboVault.
+ *
+ * @extends {module:interaction.IndirectKeystoreInteraction}
+ */
+export class CoboVaultInteraction extends IndirectKeystoreInteraction {
+}
+
+/**
+ * Base class for interactions which read a QR code displayed by a
+ * CoboVault command.
+ *
+ * @extends {module:cobovault.CoboVaultInteraction}
+ */
+
+export class CoboVaultReader extends CoboVaultInteraction {
+    constructor() {
+        super();
+        this.reader = true;
+    }
+
+    messages() {
+        const messages = super.messages();
+        messages.push({
+            state: ACTIVE,
+            level: INFO,
+            code: "cobovault.scanning",
+            text: "Scan Cobo Vault QR code now.",
+        });
+        return messages;
+    }
+
+    parse(qrs) {
+        return smartParseQR(qrs)
+    }
+}
+
+/**
+ * Base class for interactions which display data as a QR code for
+ * CoboVault to read and then read the QR code CoboVault displays in
+ * response.
+ *
+ * @extends {module:cobovault.CoboVaultInteraction}
+ */
+export class CoboVaultDisplayer extends CoboVaultInteraction {
+
+    constructor() {
+        super();
+        this.displayer = true;
+    }
+
+    encodeUR(data) {
+        return encodeUR(data, 800);
+    }
+}
+
+
+/**
+ * Reads an extended public key from data in a CoboVault QR code.
+ *
+ * @extends {module:cobovault.CoboVaultReader}
+ * @example
+ * const interaction = new CoboVaultExportExtendedPublicKey();
+ * const encodedString = readCoboVaultQRCode(); // application dependent
+ * const {xpub, bip32Path} = interaction.parse(encoodedString);
+ * console.log(xpub);
+ * // "xpub..."
+ * console.log(bip32Path);
+ * // "m/45'/0'/0'"
+ */
+
+export class CoboVaultExportExtendedPublicKey extends CoboVaultReader {
+    messages() {
+        const messages = super.messages();
+        messages.push(commandMessage({
+            instructions: "①Please open Cobo Vault, go to Menu > Multisig Wallet > More > Show/Export  XPUB. ②Click the camera icon above and scan the QR Code of XPUB displays on Cobo Vault",
```

Comment:
this might be better as multiple messages instead of one with multiple instructions inside